### PR TITLE
MOTECH-2889 Added support for CommCare's Report Data endpoint

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataColumn.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package org.motechproject.commcare.domain.report;
+
+import com.google.gson.annotations.SerializedName;
+
+/*
+*
+ * Represents column part in single report.
+*/
+
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataColumn.java]: Remove empty lines.

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataColumn.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
+package org.motechproject.commcare.domain.report;
+
+import com.google.gson.annotations.SerializedName;
+
+/*
+*
+ * Represents column part in single report.
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataColumn.java]: Represents a single column in a report data.

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataContainerJson.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+import java.util.Map;
+
+/*
+*
+ * Represents a single CommCareHQ report data. It's a part of the CommCareHQ model.
+*/
+
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataContainerJson.java]: Remove empty lines.

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package org.motechproject.commcare.domain.report;
+
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents a single CommCareHQ report data. It's a part of the MOTECH model.
+ */
+
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java]: Remove empty line.

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+public class ReportDataInfo {
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java:10]: Seems like a duplicate of ReportDataContainerJson, what is the point? I believe we got rid of the OpenMRS/MOTECH model distinction some time ago.

File: commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package org.motechproject.commcare.service;
+
+import org.motechproject.commcare.domain.report.ReportDataInfo;
+
+
+/**
+ * This service is responsible for Interacting with CommCareHQ's programmatic Report Data APIs and uploading JSON data
+ * from single report.
+ */
+
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java]: Remove empty line.

File: commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package org.motechproject.commcare.service;
+
+import org.motechproject.commcare.domain.report.ReportDataInfo;
+
+
+/**
+ * This service is responsible for Interacting with CommCareHQ's programmatic Report Data APIs and uploading JSON data
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java]: Cut the "This service is" part.

Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java]: interacting

Comment (ngraczewski) [reply commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java]: CommCareHQ's Report Data APIs

File: commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * from single report.
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/service/CommcareReportDataService.java]: a single

File: commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareReportDataServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import org.motechproject.commcare.service.CommcareConfigService;
+import org.motechproject.commcare.service.CommcareReportDataService;
+import org.motechproject.commons.api.json.MotechJsonReader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Type;
+
+
+@Service
```
Comment (ngraczewski) [inline commcare/src/main/java/org/motechproject/commcare/service/impl/CommcareReportDataServiceImpl.java]: Missing javadoc.

**Review by ngraczewski — CHANGES_REQUESTED** [review]

File: commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java (hunk 5 above)
Comment (ghost) [reply commcare/src/main/java/org/motechproject/commcare/domain/report/ReportDataInfo.java:10]: It is prepared to use exactly the same way like similarly class in Cases part in CommCare model and can be useful in future actions.

**Comment by kmadej** [comment]
please add unit tests

**Comment by jredlarski** [comment]
Closing this PR. The work is continued under [PR#457](https://github.com/motech/modules/pull/457)

**Pull request closed without merging**